src/fireedge/src/modules/containers/Hosts/Hosts.js

```diff
@@ -0,0 +1,217 @@
+ */
+export function Hosts() {
+  const [selectedRows, setSelectedRows] = useState(() => [])
+  const actions = HostsTable.Actions()
+  const { zone } = useGeneral()
+
+  return (
+    <MuiProvider theme={SunstoneTheme}>
+      <TranslateProvider>
+        <ResourcesBackButton
+          selectedRows={selectedRows}
+          setSelectedRows={setSelectedRows}
+          useUpdateMutation={HostAPI.useUpdateHostMutation}
+          zone={zone}
+          actions={actions}
+          table={(props) => (
+            <HostsTable.Table
+              onSelectedRowsChange={props.setSelectedRows}
+              globalActions={props.actions}
+              useUpdateMutation={props.useUpdateMutation}
+              onRowClick={props.resourcesBackButtonClick}
+              zoneId={props.zone}
+              initialState={{
+                selectedRowIds: props.selectedRowsTable,
+              }}
+            />
+          )}
+          simpleGroupsTags={(props) => (
+            <GroupedTags
+              tags={props.selectedRows}
+              handleElement={props.handleElement}
+              onDelete={props.handleUnselectRow}
+            />
+          )}
+          info={(props) => {
+            const propsInfo = {
+              host: props?.selectedRows?.[0]?.original,
+              selectedRows: props?.selectedRows,
+            }
+            props?.gotoPage && (propsInfo.gotoPage = props.gotoPage)
+            props?.unselect && (propsInfo.unselect = props.unselect)
+
+            return <InfoTabs {...propsInfo} />
+          }}
+        />
+      </TranslateProvider>
+    </MuiProvider>
+  )
+}
+
+/**
+ * Displays details of a Host.
+ *
+ * @param {Host} host - Host to display
+ * @param {Function} [gotoPage] - Function to navigate to a page of a Host
+ * @param {Function} [unselect] - Function to unselect a Host
+ * @returns {ReactElement} Host details
+ */
+const InfoTabs = memo(({ host, gotoPage, unselect }) => {
+  const [getVm, { data: lazyData, isFetching }] = HostAPI.useLazyGetHostQuery()
+  const id = host?.ID ?? lazyData?.ID
+
+  const { settings: { FIREEDGE: fireedge = {} } = {} } = useAuth()
+  const { FULL_SCREEN_INFO } = fireedge
+  const { fullViewMode } = SERVER_CONFIG
+  const fullModeDefault =
+    FULL_SCREEN_INFO !== undefined ? FULL_SCREEN_INFO === 'true' : fullViewMode
+  const { isFullMode } = useGeneral()
+  const { setFullMode } = useGeneralApi()
+
+  useEffect(() => {
+    !isFullMode && gotoPage()
+  }, [])
+
+  return (
+    <Stack overflow="auto">
+      <Stack
+        direction="row"
+        alignItems="center"
+        justifyContent="space-between"
+        gap={1}
+        mx={1}
+        mb={1}
+      >
+        <Stack direction="row">
+          {fullModeDefault && (
+            <SubmitButton
+              data-cy="detail-back"
+              icon={<NavArrowLeft />}
+              tooltip={Tr(T.Back)}
+              isSubmitting={isFetching}
+              onClick={() => unselect()}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using Function.prototype.bind and arrows functions as attributes will negatively impact performance in React. Each time the parent is rendered, the function will be re-created and trigger a render of the component causing excessive renders and more memory use.